[WIP] Sally please review - Everyone else don't merge - Thanks!

AliSchlereth opened the pull request:
@sallymacnicholas  Please look over what we have done so far. 
Esp the following:
Built a rake task to import the data from the CSV we made.
 -The rake task is found in lib/tasks/import.rake
 -There is a call to a class method on State since the data doesn't have state abbreviations and the d3 library we plan to use relies on the abbreviations. Is this an okay way to solve the problem? Should this method live in the rake task?

Built out an endpoint to report on Median Expenditures for each state comparing charter spending to non-charter spending.
  -Feature test
  -Route & Controller
  -ActiveRecord & some Ruby to output it as a hash (Did we do this right, how could we make it better? Should some of this be done in a serializer? So many questions)

@bermannoah @dijital2016 @tmikeschu  Do you have any further questions for Sally to review?

sallymacnicholas commented:
I think a better way of doing the state abbreviation would be to do a before_save action in the model that is actually setting the state abbreviation. I would pull out the hash from the model and put that in a module instead. This cleans up the model, and then also the before_save will allow you to set the state abbreviation anytime you save a state, which down the road could even be outside of a rake task. IE, if you ever were to allow users to create this kind of data, your model will already be setup to handle this.

I'll need to understand a little more on what MedianExpenditure is supposed to be doing and how you are planning on using the d3 library to answer your questions on that.

AliSchlereth commented:
@sallymacnicholas We've updated the logic for formatting the input.  We still have yet to pull the setting the state abbreviation out to a module.  
Will you look at the changes we've made? 
-Updated Med Exp method
-Created a PORO called FormatMedExpResponse
-Updated tests

@bermannoah

bermannoah commented:
The good branch, the best branch.
